chat_exporter/construct/message.py

```diff
@@ -241,11 +244,31 @@ async def build_sticker(self):
             ("ATTACH_URL_THUMB", str(sticker_image_url), PARSE_MODE_NONE)
         ])
 
-    async def build_assets(self):
+    async def build_assets(self, channel: Optional[discord.TextChannel]):
         for e in self.message.embeds:
             self.embeds += await Embed(e, self.guild).flow()
 
         for a in self.message.attachments:
+            if channel:
+                try:
+                    async with aiohttp.ClientSession() as session:
+                        async with session.get(a.url) as res:
+                            if res.status != 200:
+                                res.raise_for_status()
+                            data = io.BytesIO(await res.read())
+                            data.seek(0)
+                            attach = discord.File(data, a.filename)
+                            msg: discord.Message = await channel.send(file=attach)
+                            a = msg.attachments[0]
+                except discord.errors.Forbidden as e:
+                    # permission error
+                    raise e
+                except discord.errors.HTTPException as e:
+                    # general http error
+                    raise e
```

Comment:
```
except (discord.errors.Forbidden, discord.errors.HTTPException) as e:
    raise e
```

Reply:
discord.errors.Forbidden is a subclass from HTTPException. I just added it explicitly to talk about if we want to deal with them differently, but imo we should raise them all to the user, so we could even remove the try ... except completely....

Reply:
If you are just doing `discord.errors.HTTPException` which raises anyways, you can remove the `try:except`. 
Alternatively, you could customise the error return. 

I'm still yet to test this locally.